pkg/ring/kv/consul/client.go

```diff
@@ -99,21 +98,17 @@ func (c *Client) Put(ctx context.Context, key string, value interface{}) error {
 		return err
 	}
 
-	return instrument.CollectedRequest(ctx, "Put", consulRequestDuration, instrument.ErrorCode, func(ctx context.Context) error {
```

Comment:
Now that I notice, I think previous version was even wrong. We did track "Put" twice for 1 Put call. Once here, and once in `consul/metrics.go`. Am I missing anything?

Reply:
After investigating further, the consul metrics probably should remain as is. They provide lower level metrics for the client interacting with consul itself: https://github.com/hashicorp/consul/blob/master/api/kv.go

In our KV implementation CAS is retried multiple times and by using a generic metrics client we are missing out on instrumenting some of these calls. One option that may be used to solve this problem is moving retries to there own KV implementation and wrap clients with it.

Reply:
I am going to re-add the separate consul instrumentation and will add an issue to add a generic retry implementation for the KV store interface. WDYT?